core/trino-main/src/main/java/io/trino/sql/planner/planprinter/JsonRenderer.java

```diff
@@ -109,4 +119,32 @@ public String getDetails()
             return remoteSources;
         }
     }
+
+    public static class JsonPlanFragment
```

Comment:
To do this properly, we should introduce a `DistributedPlanRepresentation` that contains `PlanFragmentRepresentation`, which in turn contains `NodeRepresentation` objects.

Then the text and json renderers should be updated to be able to render a `DistributedPlanRepresentation` in addition to `PlanRepresentation`. `formatFragment` and `formatJsonFragmentList` should move from `PlanPrinter` into the respective renderers.

Reply:
Refactored `formatFragmentList` into JsonRenderer. Now, `DistributedPlanRepresentation` contains all `PlanFragmentRepresentation` of the corresponding `PlanFragments`. But instead of using `NodeRepresentation` to build the json, we are utilizing `PlanFragment::getJsonRepresentation` to get the json representation directly.